Lib/asyncio/executor.py

```diff
@@ -0,0 +1,226 @@
+import time
+from inspect import isawaitable
+from typing import (Any, AsyncIterable, Awaitable, Iterable, NamedTuple,
+                    Optional, Protocol, cast)
+
+from .exceptions import CancelledError
+from .futures import Future
+from .locks import Event
+from .queues import Queue, QueueShutDown
+from .tasks import FIRST_COMPLETED, Task, create_task, gather, wait, wait_for
+
+__all__ = (
+    "Executor",
+)
+
+
+class _WorkFunction[R](Protocol):
+    def __call__(self, *args, **kwargs) -> R | Awaitable[R]:
+        ...
```

Comment:
```suggestion
class _WorkFunction[R, **P](Protocol):
    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> R | Awaitable[R]:
        ...
```
If the module has type hints -- let's use it constantly everywhere